### ISSUE TITLE
ddt4all crashes with DCM12 when reading the DTC errors

### ISSUE THREAD

**Issue by octavp** [cedricp/ddt4all#741]
using the DCM12 -L4CB1 - v9.3, double click on the ECM to populate the screens , then pressing the read DTC errors button leads to the ddt4all crash. elm and eculogs are also attached.

![ddt4all_crash_trace_data](https://user-images.githubusercontent.com/1552731/196799430-c45cc7fb-2a62-4e51-83b4-dd11fbbeb9cb.PNG)
[ecu_ddt_snippet_02.txt](https://github.com/cedricp/ddt4all/files/9824135/ecu_ddt_snippet_02.txt)
[elm_ddt_snippet_02.log](https://github.com/cedricp/ddt4all/files/9824136/elm_ddt_snippet_02.log)

**Comment by KarelSvo** [comment]
I can not understand your problem with the ddt4all Python 2 version.
Are you using the Python 3 version?
The log says Elm doesn't give a positive response when requesting the Fast Init routine. (AT FI = ?)

**Comment by octavp** [comment]
I am using the python 3.8. what makes you think I am using python 2?

**Comment by KarelSvo** [comment]
I had tried with the Python 2 version and found no problems.

**Comment by octavp** [comment]
can you please share your logs? how did you exchange the python3.8 that comes with ddt4all release package with python 2. I tried pyren (which comes with Python 2.7) but that too was not able to open the elm (I get ELM not responding).

**Comment by octavp** [comment]
I tried starting under python 2.7 but ddt4all.py complains about importlib.machinery module. there is no such thing for 2.7.  how did you run ddt4all.py under 2.7?

**Comment by KarelSvo** [comment]
https://github.com/cedricp/ddt4all/releases/tag/1.5

**Comment by octavp** [comment]
thank you. I just realized that I should use a different release package.

**Comment by octavp** [comment]
I did the same with ddt4all using python 2.7 and with pyren (in ddt mode). pyren logs are showing BUS INIT errors (I attached the pyren logs). I attached the logs from my ddt4all tries with python 2.7 as well.
[pyren_ecu_log.txt](https://github.com/cedricp/ddt4all/files/9838056/pyren_ecu_log.txt)
[pyren_elm_log.txt](https://github.com/cedricp/ddt4all/files/9838057/pyren_elm_log.txt)
[ddt4all_python27_ecu_ddt.txt](https://github.com/cedricp/ddt4all/files/9838116/ddt4all_python27_ecu_ddt.txt)
[ddt4all_python27_elm_ddt.txt](https://github.com/cedricp/ddt4all/files/9838117/ddt4all_python27_elm_ddt.txt)

**Comment by KarelSvo** [comment]
In your log files there is no vehicle type to scan for ecu and no Elm test to watch.
Don't want to check your Elm hardware first like I recommended? https://github.com/cedricp/ddt4all/issues/739#issuecomment-1284829787 Why are you starting 2 threads about the same problem?

**Comment by octavp** [comment]
I thought there are 2 different issues. Meanwhile it seems that they are not. I used the ddt4all with ecu.zip and that one is trying to open the ELM but it fails with Windows Error 5 (Access Denied) on Windows 10. I tried a different windows 11 machine and after it worked once (I was able to connect to the ELM successfully) I started to get the same WindowsError 5 there as well. I tried different things to fix the Access Denied error but without success so far. I tried Pyren on a borrowed Mac (Catalina) and there I could not connect to the ELM as well. I will try your recommendation tomorrow.
